src/run.ts

```diff
@@ -0,0 +1,61 @@
+import { readFile } from 'fs/promises'
+import { load } from 'js-yaml'
+import { lintFiles } from './linter'
+import { getChangedFiles } from './git'
+import type { Core, ActionInputs } from './types'
+
+async function run(core: Core): Promise<void> {
+  try {
+    const inputs: ActionInputs = {
+      githubToken: core.getInput('github_token', { required: true }),
+      apiKey: core.getInput('api_key', { required: true }),
+      axeLinterUrl:
+        core.getInput('axe_linter_url') || 'https://axe-linter.deque.com'
+    }
+
+    // Remove trailing slash if present
+    inputs.axeLinterUrl = inputs.axeLinterUrl.replace(/\/$/, '')
+
+    // Get changed files
+    const changedFiles = await getChangedFiles(inputs.githubToken)
+
+    if (changedFiles.length === 0) {
+      core.info('No files to lint')
```

Comment:
nit - the only reason a user would want to see this log is if they're debugging an issue.

```suggestion
      core.debug('No files to lint')
```